log2ram

```diff
@@ -9,28 +9,52 @@ fi
 LOG_NAME='log2ram.log'
 NO_RSYNC=${USE_RSYNC#true}
 
-isSafe () {
+## @fn is_safe()
+## @brief Check if hdd log exists
+is_safe() {
     [ -d "$HDD_LOG" ] || echo "ERROR: $HDD_LOG/ doesn't exist! Can't sync."
     [ -d "$HDD_LOG" ] || exit 1
 }
 
-remountOriginal() {
-    OPTION="$1"
-    mount -o remount,"$OPTION" "$HDD_LOG"
+## @fn journald_logrotate()
+## @brief Logrotate the journal if the current RAM_LOG path is part of the journald directory
+journald_logrotate() {
+    if ! [ -x "$(command -v journalctl)" ] || ! [ "$JOURNALD_AWARE" = true ]; then
+        return 1
+    fi
+
+    if journalctl --header | grep "File path" | grep "$RAM_LOG" >/dev/null 2>&1; then
+        journalctl --rotate
+        return 0
+    else
+        return 1
+    fi
 }
 
-syncToDisk () {
-    isSafe
+## @fn sync_to_disk()
+## @brief Sync memory back to hard disk
+sync_to_disk() {
+    is_safe
+
+    optional_params=()
 
-    if [ -z "${NO_RSYNC}" -a -x "$(command -v rsync)" ]; then
-        rsync -aXv --inplace --no-whole-file --delete-after "$RAM_LOG"/ "$HDD_LOG"/ 2>&1 | tee -a "$LOG2RAM_LOG"
+    if journald_logrotate; then
+        optional_params+=("--include=journal/*/*@*.journal")
+        optional_params+=("--exclude=journal/*/*")
```

Comment:
I talked to the folks at systemd/journald. They told me to do a logrotate so that all logging entries are archived until that moment. The resulting currently used journal, which does not have an "@" in the filename, must not be copied back because it would be marked as corrupt when restoring at the next boot.